src/import.js

```diff
@@ -111,72 +111,44 @@ Sk.loadExternalLibrary = function (name) {
 /**
  * @param {string} name to look for
  * @param {string} ext extension to use (.py or .js)
- * @param {boolean=} failok will throw if not true
- * @param {boolean=} canSuspend can we suspend?
- * @param {string=} currentDir if any
+ * @param {Object=} searchPath an iterable set of path strings
  */
-Sk.importSearchPathForName = function (name, ext, failok, canSuspend, currentDir) {
+Sk.importSearchPathForName = function (name, ext, searchPath) {
     var fn;
     var j;
     var fns = [];
     var nameAsPath = name.replace(/\./g, "/");
-    var L = Sk.realsyspath;
     var it, i;
 
-    for (it = L.tp$iter(), i = it.tp$iternext(); i !== undefined; i = it.tp$iternext()) {
-        fns.push(i.v + "/" + nameAsPath + ext);                 // module
-        fns.push(i.v + "/" + nameAsPath + "/__init__" + ext);   // package
-    }
-
-    fns.push(currentDir + "/" + nameAsPath + ext);
-    fns.push(currentDir + "/" + nameAsPath + "/__init__" + ext);
-
-    j = 0;
-
-    return (function tryNextPath() {
-        var handleRead = function handleRead(s) {
-            var ns;
-            if (s instanceof Sk.misceval.Suspension) {
-                ns = new Sk.misceval.Suspension(undefined, s);
-                ns.resume = function() {
-                    try {
-                        return handleRead(s.resume());
-                    } catch (e) {
-                        j++;
-                        return tryNextPath();
-                    }
-                };
-                return ns;
-            } else {
-                return {filename: fns[j], code: s};
+    var tryPathAndBreakOnSuccess = function(filename, packagePath) {
+        return Sk.misceval.chain(
+            Sk.misceval.tryCatch(function() {
+                return Sk.read(filename);
```

Comment:
Does this mean, Sk.read can now return a suspension?

Reply:
Yes. It always could, since about 2014 (it's why the first suspension-ised version of `import` was so tangled). Only now the code is nice enough that you can notice :D